mysql/mariadb_gtid.go

```diff
@@ -182,10 +189,15 @@ func (s *MariadbGTIDSet) Encode() []byte {
 // Clone clones a mariadb gtid set
 func (s *MariadbGTIDSet) Clone() GTIDSet {
 	clone := &MariadbGTIDSet{
-		Sets: make(map[uint32]*MariadbGTID),
+		Sets: make(map[uint32]map[uint32]*MariadbGTID),
 	}
-	for domainID, gtid := range s.Sets {
-		clone.Sets[domainID] = gtid.Clone()
+	for domainID, set := range s.Sets {
+		for serverID, gtid := range set {
+			if clone.Sets[domainID] == nil {
```

Comment:
We can move the creating of domainID level map out of serverID loop (above line 195), then no need to check nil.